### PR TITLE
de-point context; fix race; init rand; tidy test pointers - and more...

### PR DESCRIPTION
**Pull request by gedge**
### What

 - add init() seeding of rand
 - ticker methods use pointer
 - fix some race conditions
 - fail/return if Unmarshal returns an error
 - tests: make status codes generic
 - docs: remove dots from methods
 - tests: add cancelled (context) healthcheck test
 - tests: new utils_test.go
 - tests: remove superfluous time.Duration() calls
 - tests: make testVersion a const, among others

### How to review

check for sanity, run tests

### Who can review

@dandwal should have a look, at least

### CODE REVIEW

File: healthcheck/utils.go [hunk 1]
```diff
@@ -5,18 +5,21 @@ import (
...
+	rand.Seed(time.Now().UnixNano())
+}
+
+// calcIntervalWithJitter returns a new duration based on a provided interval and a jitter of ±jitterFactor
 func calcIntervalWithJitter(interval time.Duration) time.Duration {
-	const jitterAmount = 0.05
-	upperJitterThreshold := int(float64(interval) * jitterAmount)
-	lowerJitterThreshold := -1 * upperJitterThreshold
-	jitterToApply := time.Duration(random(lowerJitterThreshold, upperJitterThreshold))
+	upperJitterThreshold := int64(float64(interval) * jitterFactor)
```
Comment (nshumoogum) [inline healthcheck/utils.go]: I think it is better to be explicit with defining both lower and upper jitter thresholds, but would have a called it maxJitter and minJitter.

like so:

```
fun getIntervalWithJitter(interval time.Duration) time.Duration {
  maxJitter := int(float64(interval) * jitterAmount)
  minJitter := -maxJitter
  jitter := time.Duration(random(minJitter, maxJitter)

  return interval + jitter
}
```

Comment (gedge) [reply healthcheck/utils.go]: you and Dan out-vote me on the clarity of two vars, so I'll concede this 👍

File: healthcheck/utils_test.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	"testing"
+	"time"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestJitter(t *testing.T) {
+
+	interval := 120 * time.Second
+	// this should match the calculation for upperJitterThreshold
```
[the remark this answers is not in the document]
Comment (nshumoogum) [reply healthcheck/utils_test.go]: I agree, good spot

[the remark this answers is not in the document]
Comment (gedge) [reply healthcheck/utils_test.go]: Agreed, so I've made a func to make this clearer, please re-review

**Pull request merged**